File: app/src/main/java/com/buzbuz/dikkeneksoundboard/TabPagerAdapter.java
package com.buzbuz.dikkeneksoundboard;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class TabPagerAdapter extends FragmentPagerAdapter {

    /** Number of pages in Pager */
    private static final int PAGE_COUNT = 2;
    /** Name of tabs */
    private static final String mTabTitles[] = new String[] { "Favoris", "Tous" };

    public TabPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        return SoundListFragment.newInstance(position == 0);
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTabTitles[position];
    }

}

File: app/src/main/java/com/buzbuz/dikkeneksoundboard/ui/TabPagerAdapter.kt
/*
 * Copyright (C) 2020 Nain57
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; If not, see <http://www.gnu.org/licenses/>.
 */
package com.buzbuz.dikkeneksoundboard.ui

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.buzbuz.dikkeneksoundboard.R

/** List of titles string resource identifier for the tabs. */
private val TAB_TITLES = arrayOf(
    R.string.tab_all_sounds,
    R.string.tab_favourites_sounds
)

/** Adapter for the view pager populating it with the 'all' and 'favourites' fragments. */
class TabPagerAdapter(private val context: Context, fragmentManager: FragmentManager)
    : FragmentPagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(position: Int): Fragment = SoundListFragment.newInstance(position != 0)

    override fun getPageTitle(position: Int): CharSequence? = context.resources.getString(TAB_TITLES[position])

    override fun getCount(): Int = TAB_TITLES.size
}